test/support/network-integration/collections/ansible_collections/ansible/netcommon/plugins/connection/older_plugins.py

```diff
@@ -0,0 +1,1391 @@
+      - I(libssh) will use the ansible-pylibssh package, which needs to be installed in order to work.
+      - I(paramiko) will instead use the paramiko package to manage the SSH connection.
+      - I(auto) will use ansible-pylibssh if that package is installed, otherwise will fallback to paramiko.
+    default: auto
+    choices: ["libssh", "paramiko", "auto"]
+    env:
+        - name: ANSIBLE_NETWORK_CLI_SSH_TYPE
+    ini:
+        - section: persistent_connection
+          key: ssh_type
+    vars:
+    - name: ansible_network_cli_ssh_type
+  host_key_checking:
+    description: 'Set this to "False" if you want to avoid host key checking by the underlying tools Ansible uses to connect to the host'
+    type: boolean
+    default: True
+    env:
+    - name: ANSIBLE_HOST_KEY_CHECKING
+    - name: ANSIBLE_SSH_HOST_KEY_CHECKING
+    ini:
+    - section: defaults
+      key: host_key_checking
+    - section: persistent_connection
+      key: host_key_checking
+    vars:
+    - name: ansible_host_key_checking
+    - name: ansible_ssh_host_key_checking
+  single_user_mode:
+    type: boolean
+    default: false
+    version_added: 2.0.0
+    description:
+    - This option enables caching of data fetched from the target for re-use.
+      The cache is invalidated when the target device enters configuration mode.
+    - Applicable only for platforms where this has been implemented.
+    env:
+    - name: ANSIBLE_NETWORK_SINGLE_USER_MODE
+    vars:
+    - name: ansible_network_single_user_mode
+"""
+
+import getpass
+import json
+import logging
+import os
+import re
+import signal
+import socket
+import time
+import traceback
+from functools import wraps
+from io import BytesIO
+
+from ansible.errors import AnsibleConnectionFailure, AnsibleError
+from ansible.module_utils.common.text.converters import to_bytes, to_text
+from ansible.module_utils.basic import missing_required_lib
+from ansible.module_utils.six import PY3
+from ansible.module_utils.six.moves import cPickle
+from ansible.playbook.play_context import PlayContext
+from ansible.plugins.loader import (
+    cache_loader,
+    cliconf_loader,
+    connection_loader,
+    terminal_loader,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    to_list,
+)
+from ansible_collections.ansible.netcommon.plugins.plugin_utils.connection_base import (
+    NetworkConnectionBase,
+)
+
+try:
+    from scp import SCPClient
+
+    HAS_SCP = True
+except ImportError:
+    HAS_SCP = False
+
+HAS_PYLIBSSH = False
+
+
+def ensure_connect(func):
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        if not self._connected:
+            self._connect()
+        self.update_cli_prompt_context()
+        return func(self, *args, **kwargs)
+
+    return wrapped
+
+
+class AnsibleCmdRespRecv(Exception):
+    pass
+
+
+class OldConnection(NetworkConnectionBase):
```

Comment:
The plugin loader looks for a class called `Connection`.

If you're still considering how to test this, I got the unit test working locally with `--docker` in test/units/executor/test_task_executor.py like this:

```python
from ansible.inventory.host import Host  # FIXME: move new imports to top
import ansible.plugins.loader as loader
from ansible.playbook import Playbook
from ansible.playbook.play_context import PlayContext
import pytest


def test_legacy_networking_connection(tmpdir, mocker):
    tmpdir.mkdir("connection_plugins").join("legacy_connection.py").write("""
from ansible.plugins.connection import NetworkConnectionBase

class Connection(NetworkConnectionBase):
    def _connect(self):
        pass

    @property
    def transport(self):
        pass

    def get_option_and_origin(self, option, hostvars=None):
        if option == 'persistent_log_messages':
            return ('fromini', False)
        elif option == 'persistent_command_timeout':
            return ('fromini', 10)
        return super().get_option_and_origin(option, hostvars)
""")

    pb = """
- hosts: all
  connection: legacy_connection
  gather_facts: false
  tasks:
  - ping:
"""

    loader.init_plugin_loader()
    loader.add_all_plugin_dirs(tmpdir)

    mock_data_loader = DictDataLoader({'playbook.yml': pb})

    mock_var_manager = MagicMock()
    mock_var_manager.get_delegated_vars_and_hostname.return_value = {}, None

    playbook_ds = Playbook.load('playbook.yml', loader=mock_data_loader, variable_manager=mock_var_manager)
    play = playbook_ds._entries[0]

    te = TaskExecutor(
        host=Host('dummy'),
        task=play.tasks[0].block[0],
        job_vars={},
        play_context=PlayContext(play=play),
        new_stdin=None,
        loader=mock_data_loader,
        shared_loader_obj=loader,
        final_q=None,
        variable_manager=mock_var_manager
    )

    # test loading the connection plugin but short-circuit actually establishing a connection
    mocker.patch('ansible.executor.task_executor.start_connection', MagicMock(side_effect=NotImplementedError('success')))
    with pytest.raises(NotImplementedError, match='success'):
        te._execute()
```